lib/util.js

```diff
@@ -186,19 +195,30 @@ function inspect(obj, opts) {
     exports._extend(ctx, opts);
   }
   // set default options
-  if (ctx.showHidden === undefined) ctx.showHidden = false;
-  if (ctx.depth === undefined) ctx.depth = 2;
-  if (ctx.colors === undefined) ctx.colors = false;
-  if (ctx.customInspect === undefined) ctx.customInspect = true;
-  if (ctx.showProxy === undefined) ctx.showProxy = false;
+  Object.keys(inspectOptions).forEach(function(option) {
+    if (ctx[option] === undefined) ctx[option] = inspectOptions[option];
+  });
   if (ctx.colors) ctx.stylize = stylizeWithColor;
-  if (ctx.maxArrayLength === undefined) ctx.maxArrayLength = kDefaultMaxLength;
   if (ctx.maxArrayLength === null) ctx.maxArrayLength = Infinity;
-  if (ctx.breakLength === undefined) ctx.breakLength = 60;
   return formatValue(ctx, obj, ctx.depth);
 }
 exports.inspect = inspect;
 
+inspect.config = function (options = {}) {
+  if (typeof options !== 'object') {
+    throw new TypeError('"options" argument must be a object');
+  }
+
+  // copy properties that exist on `inspectOptions`
+  Object.keys(options).forEach(function(opt) {
```

Comment:
You could probably get away with `Object.assign()` or `util._extend()` here. If there are unsupported options in `options`, they'll just be ignored later anyway, right?